js/packages/openinference-instrumentation-langchain/src/instrumentation.ts

```diff
@@ -0,0 +1,112 @@
+import * as CallbackManagerModule from "@langchain/core/callbacks/manager";
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+} from "@opentelemetry/instrumentation";
+import { VERSION } from "./version";
+import { diag } from "@opentelemetry/api";
+import { LangChainTracer } from "./tracer";
+
+const MODULE_NAME = "@langchain/core/callbacks";
+
+export class LangChainInstrumentation extends InstrumentationBase<
+  typeof CallbackManagerModule
+> {
+  constructor(config?: InstrumentationConfig) {
+    super(
+      "@arizeai/openinference-instrumentation-langchain",
+      VERSION,
+      Object.assign({}, config),
+    );
+  }
+
+  protected init(): InstrumentationModuleDefinition<
+    typeof CallbackManagerModule
+  > {
+    const module = new InstrumentationNodeModuleDefinition<
+      typeof CallbackManagerModule
+    >(
+      "@langchain/core/dist/callbacks/manager.cjs",
+      ["^0.0.0", "^0.1.0"],
+      this.patch.bind(this),
+      this.unpatch.bind(this),
+    );
+
+    return module;
+  }
+
+  private patch(
+    module: typeof CallbackManagerModule & {
+      openInferencePatched?: boolean;
+    },
+    moduleVersion?: string,
+  ) {
+    diag.debug(
+      `Applying patch for ${MODULE_NAME}${
+        moduleVersion != null ? `@${moduleVersion}` : ""
+      }`,
+    );
+    if (module?.openInferencePatched) {
+      return module;
+    }
+
+    this._wrap(module.CallbackManager, "configure", (original) => {
+      return (...args: Parameters<typeof original>) => {
+        const inheritableHandlers = args[0];
+        const newInheritableHandlers =
+          this._addTracerToHandlers(inheritableHandlers);
+        args[0] = newInheritableHandlers;
+        const manager = original.apply(this, args);
+        return manager;
+      };
+    });
+    module.openInferencePatched = true;
+    return module;
+  }
+
+  private unpatch(
+    module?: typeof CallbackManagerModule & {
+      openInferencePatched?: boolean;
+    },
+    moduleVersion?: string,
+  ) {
+    if (module == null) {
+      return;
+    }
+    diag.debug(
+      `Removing patch for ${MODULE_NAME}${
+        moduleVersion != null ? `@${moduleVersion}` : ""
+      }`,
+    );
+    if (isWrapped(module.CallbackManager.configure)) {
+      this._unwrap(module.CallbackManager, "configure");
+    }
+    delete module.openInferencePatched;
+    return module;
+  }
+
+  private _addTracerToHandlers(handlers?: CallbackManagerModule.Callbacks) {
+    if (handlers == null) {
+      return [new LangChainTracer(this.tracer)];
+    }
+    if (handlers instanceof CallbackManagerModule.CallbackManager) {
+      if (
+        handlers.inheritableHandlers.some(
+          (handler) => handler instanceof LangChainTracer,
+        )
+      ) {
```

Comment:
does this actually hit? also you can write this so that it's more readable
```suggestion
      tracer_already__registered = handlers.inheritableHandlers.some(
          (handler) => handler instanceof LangChainTracer,
        );
      if (
        tracer_already__registered
      ) {
```

Reply:
I believe it can when a callback manager gets copied which is apart of their functionality